ufl/adjoint.py

```diff
@@ -61,6 +62,15 @@ def __new__(cls, *args, **kw):
             # reconstruction, making it a robust strategy for handling
             # subclasses of `ufl.Coargument`.
             return primal_arg
+        elif isinstance(form, Interpolate):
+            from ufl.algorithms.replace import replace
+
+            v, u = form.arguments()
+            mapping = {
+                v: type(v)(v.ufl_function_space(), number=1, part=v.part()),
+                u: type(u)(u.ufl_function_space(), number=0, part=u.part()),
+            }
+            return replace(form, mapping)
```

Comment:
We have been allowing `form = Interpolate(Argument(V1, 0), Coargument(V2, 0))` with both arguments with `number() == 0`. 

As a consequence, `Adjoint(form)` gets the numbering wrong, but this simplification seems to do the right thing, as long as `v, u = form.arguments()` come in the right order.

Reply:
I think I solved this issue by eagerly simplifying the `Action` of an `Interpolate` with a `BaseForm`, preventing as `Interpolate` with an `Argument` that eventually gets contracted.